.travis.yml

```diff
@@ -1,26 +1,58 @@
 language: cpp
+compiler: gcc
+os: linux
+dist: bionic
+cache: ccache
 
-compiler:
-  - clang
-  - gcc
+addons:
+    apt:
+        sources:
+            - ubuntu-toolchain-r-test
+        packages:
+            - g++-4.8
+            - cmake
+            - ninja-build
+            - ccache
+            - libx11-dev
+            - libxcursor-dev
 
 env:
-  matrix:
-    - SHARED=OFF
-    - SHARED=ON
+  - SHARED=OFF
 
 before_install:
-  - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
-  - sudo apt-get update -qq
   - if [ "$SHARED" == "ON" ]; then sudo apt-get install -y libcurl4-openssl-dev libgif-dev libfreetype6-dev libjpeg-dev libz-dev libpng-dev libtinyxml-dev libpixman-1-dev liballegro4.2-dev ; fi
-  - sudo apt-get install -y -qq g++-4.8
-  - if [ "$CXX" = "g++" ]; then export CXX="g++-4.8"; fi
+  - if [ "${TRAVIS_OS_NAME}" = "linux" ] && [ "$CXX" = "g++" ]; then export CXX="g++-4.8"; fi
```

Comment:
Not necessarily something to change in this PR, but we should use a newer GCC eventually. Latest would be best for performance, but alternatively Bionic's default version might be a good start.